corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js

```diff
@@ -314,8 +316,43 @@ function DateTimeEntryBase(question, options) {
                     return;
                 }
                 self.answer(moment(newDate).format(self.serverFormat));
-            }
-        });
+            },
+            onGenerate: function () {
```

Comment:
maybe just a sentence on what this function does. not super easy to tell from the code